tests/test_model_saving.py

```diff
@@ -149,6 +174,36 @@ def test_model_saving_to_pre_created_h5py_file():
     assert_allclose(out, out2, atol=1e-05)
 
 
+def test_model_saving_to_pre_created_h5py_file_with_sparse_input():
+    inputs = Input(shape=(3,), sparse=True)
+    x = Dense(2)(inputs)
+    outputs = Dense(3)(x)
+
+    model = Model(inputs, outputs)
+    model.compile(loss=losses.MSE,
+                  optimizer=optimizers.Adam(),
+                  metrics=[metrics.categorical_accuracy])
+    x = rand(1, 3, density=0.2, format='csr')
+    y = rand(1, 3, density=0.2, format='csr')
+    model.train_on_batch(x, y)
+
+    out = model.predict(x)
+    _, fname = tempfile.mkstemp('.h5')
+    with h5py.File(fname, mode='r+') as h5file:
```

Comment:
what the difference between this section and `test_functional_model_saving_with_sparse`

Reply:
we are using a pre-created h5 file here`test_model_saving_to_pre_created_h5py...`